### PR TITLE
[ENH] Test code for code in test directories

### PR DESCRIPTION
**Pull request by TonyBagnall**
Adds tests for test directory for untested test code

### CODE REVIEW

File: aeon/classification/tests/test_all_classifiers.py [hunk 1]
```diff
@@ -78,9 +85,9 @@ def test_classifier_against_expected_results(self, estimator_class):
         if classname == "HIVECOTEV2" and os.environ.get("NUMBA_DISABLE_JIT") == "1":
             return None
 
-        for data_name, data_dict, data_loader, data_seed in [
-            ["UnitTest", unit_test_proba, load_unit_test, 0],
-            ["BasicMotions", basic_motions_proba, load_basic_motions, 4],
+        for data_dict, data_loader, data_seed in [
+            [unit_test_proba, load_unit_test, 0],
+            [basic_motions_proba, load_basic_motions, 4],
```
Comment (MatthewMiddlehurst) [inline aeon/classification/tests/test_all_classifiers.py:90]: is this broken? it just gives a more informative error I'm pretty sure

**Review by MatthewMiddlehurst — COMMENTED** [review]
It might be better to just remove these files from the coverage i.e. 

https://github.com/aeon-toolkit/aeon/blob/main/.codecov.yml#L25

File: aeon/classification/tests/test_all_classifiers.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/classification/tests/test_all_classifiers.py:90]: fine by me, but I think they should then move out of test directories, only really want test code there. Maybe into utils/_testing?

Comment (TonyBagnall) [reply aeon/classification/tests/test_all_classifiers.py:90]: put them all in one directory for results recreation and exclude it?

Comment (MatthewMiddlehurst) [reply aeon/classification/tests/test_all_classifiers.py:90]: I can put something together for it

Comment (TonyBagnall) [reply aeon/classification/tests/test_all_classifiers.py:90]: I'll see if there is anything I want to salvage from this PR

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**